database/manager.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"go.uber.org/zap"
+)
+
+//
+// DatabaseManager writes all committed keys to the database on a timer.
+//
+
+type DatabaseManager struct {
+	logger      logging.Logger
+	keyManagers map[RelayerID]*keyManager
```

Comment:
Do we plan to allow additional keyManagers to be added dynamically? It doesn't look like we do right now, but if we want to add that later, we will need to add a mutex in DatabaseManager.

Reply:
No, the set of `keyManagers` is fully specified by the configuration (as is the case with the set of application relayers that interact with the `keyManagers`)